hack/tools/kapinger/clients/http.go

```diff
@@ -80,22 +80,41 @@ func NewKapingerHTTPClient(clientset *kubernetes.Clientset, labelselector string
 
 func (k *KapingerHTTPClient) MakeRequests(ctx context.Context) error {
 	ticker := time.NewTicker(k.interval)
+	sem := make(chan struct{}, 5) // semaphore to limit the number of active goroutines
+
+	serviceController, err := newServiceController(k.clientset, k.labelselector)
+	if err != nil {
+		return fmt.Errorf("error creating service controller: %w", err)
+	}
+
+	go serviceController.run(ctx)
+
 	for {
 		select {
 		case <-ctx.Done():
 			log.Printf("HTTP client context done")
 			return nil
 		case <-ticker.C:
+			if len(sem) >= cap(sem) {
+				log.Printf("max http clients reached, skipping interval")
+				continue
+			}
 			go func() {
+				sem <- struct{}{}
+				defer func() { <-sem }()
+
 				for i := 0; i < k.volume; i++ {
-					for _, url := range k.urls {
```

Comment:
Are we killing the `TARGET_TYPE=pod` functionality? I'd remove any dead code or add comments to it to fix later
https://github.com/microsoft/retina/blob/d1378d10c3bc20599475cd505ce3b970ccc4160c/hack/tools/kapinger/clients/http.go#L61-L73